src/beman/optional26/tests/optional.t.cpp

```diff
@@ -596,6 +596,21 @@ TEST(OptionalTest, RangeTest) {
     }
 }
 
+TEST(OptionalTest, HashTest) {
+    beman::optional26::optional<int> o1 = beman::optional26::nullopt;
+    beman::optional26::optional<int> o2 = beman::optional26::nullopt;
+    beman::optional26::optional<int> o3 = 42;
+    beman::optional26::optional<int> o4 = 42;
+
+    auto h1 = std::hash<beman::optional26::optional<int>>{}(o1);
+    auto h2 = std::hash<beman::optional26::optional<int>>{}(o2);
+    auto h3 = std::hash<beman::optional26::optional<int>>{}(o3);
+    auto h4 = std::hash<beman::optional26::optional<int>>{}(o4);
+
+    EXPECT_EQ(h1, h2);
+    EXPECT_EQ(h3, h4);
```

Comment:
I would also add `EXPECT_NEQ(h1, h3)` in this test and creata new one where I would expect that hashing of a series of number inside optional gives different hash values - e.g. 1, 2, 4, 8, 16, ...